File: scryfall/set.go
package scryfall

import "time"

type Set struct {
	ID            string  `json:"id"`
	Code          string  `json:"code"`
	MTGOCode      string  `json:"mtgo_code,omitempty"`
	TCGPlayerID   int     `json:"tcgplayer_id,omitempty"`
	Name          string  `json:"name"`
	SetType       SetType `json:"set_type"`
	ReleasedAt    string  `json:"released_at,omitempty"`
	BlockCode     string  `json:"block_code,omitempty"`
	Block         string  `json:"block,omitempty"`
	ParentSetCode string  `json:"parent_set_code"`
	CardCount     int     `json:"card_count"`
	Digital       bool    `json:"digital"`
	FoilOnly      bool    `json:"foil_only"`
	ScryfallURI   string  `json:"scryfall_uri"`
	URI           string  `json:"uri"`
	IconSVGURI    string  `json:"icon_svg_uri"`
	SearchURI     string  `json:"search_uri"`
}

func (s *Set) ReleasedAtTime() time.Time {
	return TimestampToTime(s.ReleasedAt)
}

type SetType string

const (
	SetTypeCore            SetType = "core"
	SetTypeExpansion               = "expansion"
	SetTypeMasters                 = "masters"
	SetTypeMasterpiece             = "masterpiece"
	SetTypeFromTheVault            = "from_the_vault"
	SetTypeSpellbook               = "spellbook"
	SetTypePremiumDeck             = "premium_deck"
	SetTypeDuelDeck                = "duel_deck"
	SetTypeDraftInnovation         = "draft_innovation"
	SetTypeTreasureChest           = "treasure_chest"
	SetTypeCommander               = "commander"
	SetTypePlanechase              = "planechase"
	SetTypeArchenemy               = "archenemy"
	SetTypeVanguard                = "vanguard"
	SetTypeFunny                   = "funny"
	SetTypeStarter                 = "starter"
	SetTypeBox                     = "box"
	SetTypePromo                   = "promo"
	SetTypeToken                   = "token"
	SetTypeMemorabilia             = "memorabilia"
)

File: packages/scryfall/src/types/set.ts
import {ScryfallDate, UUID, URI} from './scryfall';

export type Set = {
    object: 'set';
    id: UUID;
    code: string;
    mtgo_code?: string;
    tcgpayer_id?: number;
    name: string;
    set_type: SetType;
    released_at?: ScryfallDate;
    block_code?: string;
    block?: string;
    parent_set_code?: string;
    card_count: number;
    printed_size?: number;
    digital: boolean;
    foil_only: boolean;
    nonfoil_only: boolean;
    scryfall_uri: URI;
    uri: URI;
    icon_svg_uri: URI;
    search_uri: URI;
};

export function isSet(o: unknown): o is Set {
    return Boolean(o && typeof o === 'object' && (o as any).object === 'set');
}

export enum SetType {
    Core = 'core',
    Expansion = 'expansion',
    Masters = 'masters',
    Alchemy = 'alchemy',
    Masterpiece = 'masterpiece',
    Arsenal = 'arsenal',
    FromTheVault = 'from_the_vault',
    Spellbook = 'spellbook',
    PremiumDeck = 'premium_deck',
    DuelDeck = 'duel_deck',
    DraftInnovation = 'draft_innovation',
    TreasureChest = 'treasure_chest',
    Commander = 'commander',
    Planechase = 'planechase',
    Archenemy = 'archenemy',
    Vanguard = 'vanguard',
    Funny = 'funny',
    Starter = 'starter',
    Box = 'box',
    Promo = 'promo',
    Token = 'token',
    Memorabilia = 'memorabilia',
}
